save-preprocessor/src/main/kotlin/org/cqfn/save/preprocessor/controllers/DownloadProjectController.kt

```diff
@@ -230,8 +232,31 @@ class DownloadProjectController(private val configProperties: ConfigProperties,
                     log.error("Error to update test with url=$testSuiteUrl, path=$testSuitePaths")
                 }
                 .collect(Collectors.toList())
-                .subscribe()
-        }
+        }.collectList()
+            .flatMap {
+                deleteOldStandardTestSuites(newTestSuites)
+            }.subscribe()
+    }
+
+    @Suppress("TYPE_ALIAS")
+    private fun deleteOldStandardTestSuites(newTestSuites: MutableList<TestSuiteDto>): Mono<ResponseEntity<Void>> {
+        lateinit var suitesToDelete: List<TestSuiteDto>
+        return webClientBackend.get()
+            .uri("/allStandardTestSuites")
+            .retrieve()
+            .bodyToMono<List<TestSuiteDto>>()
+            .flatMap { existingSuites ->
+                suitesToDelete = existingSuites.filter { it !in newTestSuites }
+                Mono.just(ResponseEntity<Void>(HttpStatus.OK))
+            }
+            .flatMap {
```

Comment:
```suggestion
            .map { existingSuites ->
                existingSuites.filter { it !in newTestSuites }
            }
            .flatMap { suitesToDelete ->
```
Why do you need a separate variable for `suitesToDelete`?

Reply:
Yes, thank you, fixed